app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -92,69 +107,183 @@ fun KeyboardScreen(
-                Row {
-                    row.forEach { key ->
-                        Column {
-                            KeyboardKey(
-                                key = key,
-                                lastAction = lastAction,
-                                keySize = settings?.keySize ?: DEFAULT_KEY_SIZE,
-                                autoCapitalize = autoCapitalize,
-                                spacebarMultiTaps = spacebarMultiTaps,
-                                keyBorders = keyBorders,
-                                vibrateOnTap = vibrateOnTap,
-                                soundOnTap = soundOnTap,
-                                hideLetters = hideLetters,
-                                hideSymbols = hideSymbols,
-                                capsLock = capsLock,
-                                animationSpeed = settings?.animationSpeed
-                                    ?: DEFAULT_ANIMATION_SPEED,
-                                animationHelperSpeed = settings?.animationHelperSpeed
-                                    ?: DEFAULT_ANIMATION_HELPER_SPEED,
-                                minSwipeLength = settings?.minSwipeLength ?: DEFAULT_MIN_SWIPE_LENGTH,
-                                slideSensitivity = settings?.slideSensitivity ?: DEFAULT_SLIDE_SENSITIVITY,
-                                slideEnabled = slideEnabled,
-                                onToggleShiftMode = { enable ->
-                                    mode = if (enable) {
-                                        KeyboardMode.SHIFTED
-                                    } else {
-                                        capsLock = false
-                                        KeyboardMode.MAIN
-                                    }
-                                },
-                                onToggleNumericMode = { enable ->
-                                    mode = if (enable) {
-                                        KeyboardMode.NUMERIC
-                                    } else {
-                                        capsLock = false
-                                        KeyboardMode.MAIN
+            Box(
+                modifier = Modifier.weight(1f), // Take up available space equally
+            ) {
+                AndroidView(
+                    // Write the emoji to our text box when we tap one.
+                    factory = { context ->
+                        val emojiPicker = EmojiPickerView(context)
+                        emojiPicker.setOnEmojiPickedListener {
+                            ctx.currentInputConnection.commitText(
+                                it.emoji,
+                                1,
+                            )
+                        }
+
+                        emojiPicker
+                    },
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(keyboardHeight),
```

Comment:
Add `.background(MaterialTheme.colorScheme.background` here. Otherwise its transparent and doesn't look right on light or dark themes.

Reply:
I still can't figure out how to change the text color for the emoji categories, so was thinking of setting the background to light grey and ignoring dark/light themes.
theres a .setBackgroundColor() option to change the background color, but nothing for the text. .foreground will paint a single color over all of the keyboard, hiding everything underneath it, which is useless to us.

Reply:
I tried a few things, but overall it doesn't look too bad on dark theme anyway, they use a gray / orange for the foreground text color that looks okay on both light and dark.